src/grandorgue/sound/scheduler/GOSoundGroupWorkItem.cpp

```diff
@@ -144,7 +144,7 @@ void GOSoundGroupWorkItem::Finish(bool stop, GOSoundThread *pThread) {
     GOMutexLocker locker(
       m_Mutex, false, "GOSoundGroupWorkItem::Finish", pThread);
 
-    if (locker.IsLocked() && m_Done != 3)
+    while (locker.IsLocked() && m_Done != 3)
```

Comment:
```suggestion
    while (locker.IsLocked() && m_Done != 3 && !pThread->ShouldStop())
```

because GOCondition::WaitOrStop() may return when the condition is not met but the thread is marked for stopping. Your code will never finish in this case.

Reply:
Thanks. I extended the condition, but I had to add an additional check, whether `pthread` equals `nullptr`. This can be the case if `GOSoundOutputWorkItem::Exec()` is executed.

Is there a similar edgecase which must be considered for the `while` loop change in `GOSound.cpp` (https://github.com/GrandOrgue/grandorgue/pull/1607/files#diff-bea4aa67494cafccf86c6eca3b365a7f6c6d677b9ebd7fcac50748e5ce5204f4R322)?

Reply:
> Thanks. I extended the condition, but I had to add an additional check, whether `pthread` equals `nullptr`. This can be the case if `GOSoundOutputWorkItem::Exec()` is executed.

Yes, checking for pthread would also be useful.

> Is there a similar edgecase which must be considered for the `while` loop change in `GOSound.cpp` (https://github.com/GrandOrgue/grandorgue/pull/1607/files#diff-bea4aa67494cafccf86c6eca3b365a7f6c6d677b9ebd7fcac50748e5ce5204f4R322)?

Yes, these changes are welcome.